ERCS/erc-7902.md

```diff
@@ -0,0 +1,206 @@
+
+## Specification
+
+All actions in Account Abstraction within the context of EIP-5792 must be done on a single chain and atomically.
+
+We define the following list of new "capabilities" which together cover many features necessary for Account Abstraction.
+Note that use of Paymasters managed by a "paymaster web service" is described in [ERC-7677](./eip-7677).
+
+### Create [EIP-7702](./eip-7702) Authorization Capability
+
+This capability is designed to be used with [EIP-7702](./eip-7702) and requests the Wallet Application to provide
+an EIP-7702 authorization tuple for the specified address as part of the AA transaction.
+
+Identifier:
+
+`eip7702Auth`
+
+Interface:
+
+```typescript
+type SetCodeForEOACapabilityParams = Record<
+  {
+    account: `0x${string}`,       // EOA address
+    delegation: `0x${string}`,    // delegation address
+  }
+>
+```
+
+Supporting Wallet Applications MUST generate an EIP-7702 compatible transaction that sets a code of the `account` EOA address
+to the code of `delegation` specified in the request.
+
+### Static Paymaster Configuration Capability
+
+The purpose of this capability is allowing applications to integrate with Paymasters that do not require
+the Wallet Application to resolve any dynamic configuration.
+
+The application may hard-code or resolve these parameters first and pass them with this capability.
+
+Identifier:
+
+`staticPaymasterConfiguration`
+
+Interface:
+
+```typescript
+type StaticPaymasterConfigurationCapabilityParams = Record<
+  {
+    paymaster: string;
+    paymasterData: string;
+    paymasterValidationGasLimit: `0x${string}`;
+    paymasterPostOpGasLimit: `0x${string}`;
+  }
+>;
+```
+
+### Validity Time Range Capability
+
+The purpose of this capability is allowing the applications to explicitly specify the time range during which
+the requested operations will be valid after signing.
+
+Identifier:
+
+`validityTimeRange`
+
+Interface:
+
+```typescript
+type ValidityTimeRangeCapabilityParams = Record<
+  {
+    validAfter: `0x${string}`, // operation valid only after this timestamp, in seconds
+    validUntil: `0x${string}`  // operation valid only before this timestamp, in seconds
+  }
+>
+```
+
+The Wallet Application MUST verify the time range [`validAfter`..`validUntil`] is valid and present it to the
+user in a human-readable way for confirmation as part of the transaction information.
+
+The Smart Contract Account MUST specify the time range [`validAfter`..`validUntil`] as the transaction validity range.
+
+### Multidimensional Nonce Capability
+
+The purpose of this capability is allowing the applications to explicitly specify the components of the
+semi-abstracted nonce as defined in [ERC-4337](./eip-4337.md).
+
+Identifier:
+
+`multiDimensionalNonce`
+
+Interface:
+
+```typescript
+type MultiDimensionalNonceCapabilityParams = Record<
+  {
+    nonceKey: `0x${string}`,
+    nonceSequence: `0x${string}`
```

Comment:
nonceSequence is optional. the 
- If nonceSequence is missing, the wallet should use the next nonceKey.
- explicitly specifying nonceSequence=0 means that the nonceKey is unique.
- other explicit values are discouraged, since it requires the app to know how to query the current nonce - there is a `getNonce(key)` in SCAs, but it is not a standard.